### ISSUE TITLE
Reverting two levels deep with ForeignKeys

### ISSUE THREAD

**Issue by thongly** [etianen/django-reversion#405]
**From within the admin**, I can revert with a single `ForeignKey` relationships, however, I was wondering if it's possible to do so where there are two levels of `ForeignKey`'s, but **still from within the admin**.

For instance, based on the pseudocode below:

```
class Book(models.Model):
    name = CharField(... ...)

reversion.register(Book, follow=['chapters'])

class Chapter(models.Model):
    book = ForeignKey(Book, related_name='chapters')
    name = CharField(... ...)

reversion.register(Chapter, follow=['pages'])

class Page(models.Model):
    chapter = ForeignKey(Chapter, related_name='pages')
    num = IntegerField(... ...)

reversion.register(Page)
```

**From within the admin:**
- [x] I **can** delete and revert a `Book` instance, fully recovering its underlying `Chapter` instances
- [x] I **can** delete and revert a `Chapter` instance, fully recovering its underlying `Page` instances
- [ ] I would like to, but **can NOT** delete a `Book` instance, fully recovering both its underlying `Chapter` instances, as well as the underlying `Page` instances under each of the `Chapter`'s
        
Is this possible? If not, would you be open to a pull request with this feature?

**Comment by etianen** [comment]
Using the low-level API, this should be very possible. Are you trying to get this to work with the admin integration?

**Comment by thongly** [comment]
Yes. I can do this with the API. I'm referring to the admin.

**Comment by etianen** [comment]
I think that the deeper issue here is that the django admin does not support double-nested inline models. Until it does, getting this functionality into django-reversion is going to be extremely difficult, if not impossible.

**Comment by thongly** [comment]
I'm wondering whether it's possible to fire off a custom action that in turn calls the low-level API, rather than relying on the default admin integration.

**Comment by etianen** [comment]
The way to do it is probably throw away the current form-based approach, and have some custom UI for displaying the previous version to the user. That will prevent the user from editing the previous revision before saving. Then, like you say, use the low-level API.

It’s not something I’d really want to support in django-reversion until the Django admin supports arbitrary nesting. You can probably implement this for your own purposes, however.

**Comment by jmurty** [comment]
The "strict" revert feature in PR #424 might help with this case, in that it allows you to do a more low-level revert within the admin integration if you set the `REVERSION_ADMIN_STRICT_REVERT` option.

It won't help with displaying double-nested inlines in the admin, but if it is okay for you to show just a subset of the data in the admin prior to performing a full revert, this might be enough.

**Comment by etianen** [comment]
This should be fixed in master with aa858f8ff53b50ea10be8b2ccca96ca993b95545. Try it out!
